core/core.go

```diff
@@ -1,189 +1,69 @@
-	GeneratePNG(code string) ([]byte, error)
-}
-
-type client struct {
-	options options
-
-	baseURL string
-}
-
-func (c *client) GenerateSVG(code string) ([]byte, error) {
-	p, err := code2Path(code)
-	if err != nil {
-		return nil, err
-	}
-	return c.requestHandler("svg/" + p)
+// Node diagram's definition node.
+type Node struct {
+	ID         string `json:"id"`
+	Label      string `json:"label"`
+	Group      string `json:"group"`
+	Technology string `json:"technology"`
+	External   bool   `json:"external"`
+	IsQueue    bool   `json:"is_queue"`
+	IsDatabase bool   `json:"is_database"`
 }
 
-func (c *client) GeneratePNG(code string) ([]byte, error) {
-	p, err := code2Path(code)
-	if err != nil {
-		return nil, err
-	}
-	return c.requestHandler("png/" + p)
-}
-
-func (c *client) requestHandler(p string) ([]byte, error) {
-	resp, err := c.options.httpClient.Get(c.baseURL + p)
-	if err != nil {
-		return nil, err
-	}
-
-	if resp.StatusCode > 209 {
-		return nil, errors.New("error status code: " + strconv.Itoa(resp.StatusCode))
-	}
-
-	buf, err := io.ReadAll(resp.Body)
-	defer func() { _ = resp.Body.Close() }()
-	if err != nil {
-		return nil, err
-	}
-
-	return buf, nil
+// Link diagram's definition link.
+type Link struct {
+	From       string `json:"from"`
+	To         string `json:"to"`
+	Direction  string `json:"direction"`
+	Label      string `json:"label"`
+	Technology string `json:"technology"`
 }
 
-const (
-	baseURL        = "https://www.plantuml.com/plantuml/"
-	defaultTimeout = 1 * time.Minute
-)
-
-// NewPlantUMLClient initiates the client to communicate with the plantuml server.
-func NewPlantUMLClient(optFns ...func(*options)) PlantUMLClient {
-	o := options{
-		httpClient: nil,
-	}
-
-	for _, fn := range optFns {
-		fn(&o)
-	}
-
-	resolveHTTPClient(&o)
-
-	return &client{
-		options: o,
-		baseURL: baseURL,
-	}
+// ResponseDiagram response object.
+type ResponseDiagram interface {
+	// ToJSON encodes the result as JSON.
+	ToJSON() []byte
 }
 
-func resolveHTTPClient(o *options) {
-	if o.httpClient == nil {
-		o.httpClient = &http.Client{Timeout: defaultTimeout}
-	}
+// ResponseC4Diagram resulting C4 diagram.
+type ResponseC4Diagram struct {
+	SVG string `json:"svg"`
 }
 
-// code2Path converts the diagram as code to the 64Bytes encoded string to query plantuml
-//
-// Example: the diagram's code
-// @startuml
-//
-//	a -> b
-//
-// @enduml
-//
-// will be converted to SoWkIImgAStDuL80WaG5NJk592w7rBmKe100
-//
-// The resulting string to be used to generate C4 diagram
-// - as png: GET www.plantuml.com/plantuml/png/SoWkIImgAStDuL80WaG5NJk592w7rBmKe100
-// - as svg: GET www.plantuml.com/plantuml/svg/SoWkIImgAStDuL80WaG5NJk592w7rBmKe100
-func code2Path(s string) (string, error) {
-	zb, err := compress(*(*[]byte)(unsafe.Pointer(&s)))
+func (r ResponseC4Diagram) ToJSON() []byte {
+	// FIXME(?): add svg validation.
+	o, err := json.Marshal(r)
 	if err != nil {
-		return "", err
+		panic(err)
```

Comment:
**suggestion** we should avoid panic and also IMO, we don't need this ToJSON function just to wrap the `json` usage, or do you have some reason for doing that?

Reply:
The reason for `ToJSON` method: response is an interface. I was on a fence between "Serialize" and "ToJSON" - went with the latest for the sake of clarify.
You are right about panic, thanks! I accept the feedback and implement the change ;)

Reply:
I adjusted the signature. The interface shall implement the `MustMarshal` method now to be consistent with the panic-convention.